lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -589,6 +589,21 @@ defmodule Kernel.WarningTest do
 
     assert output =~ "key :a will be overridden in map"
     assert output =~ "key 1 will be overridden in map"
+
+    pid = start_supervised!({Agent, fn -> 1 end})
+
+    next = fn ->
+      Agent.get_and_update(pid, fn acc -> {acc, acc + 1} end)
+    end
+
+    output =
+      capture_err(fn ->
+        defmodule NonLiteralDuplicateMapKeys do
+          assert %{next.() => 1, next.() => 2} == %{1 => 1, 2 => 2}
```

Comment:
Nitpick: You can use `System.unique_integer()`, so no need for the agent. Also, you don't have to wrap this in a module, just write the code, without captyre_log and all. If it warns, we will know when we run the suite. :)

So I would do something like this:

    assert map_size(%{System.unique_integer() => 1, System.unique_integer() => 2}) == 2

Reply:
I didn't like the `Agent` approach, but it was the only idea that I had. I like your idea!